php/class-relate.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Relate class for the Cloudinary plugin.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary;
+
+use Cloudinary\Connect\Api;
+use Cloudinary\Relate\Relationship;
+
+/**
+ * Class Relate
+ */
+class Relate {
+
+	/**
+	 * Holds the plugin instance.
+	 *
+	 * @var Plugin Instance of the global plugin.
+	 */
+	public $plugin;
+
+	/**
+	 * Holds the Media instance.
+	 *
+	 * @var Media
+	 */
+	protected $media;
+
+	/**
+	 * Relate constructor.
+	 *
+	 * @param Plugin $plugin Instance of the main plugin.
+	 */
+	public function __construct( $plugin ) {
+		$this->plugin = $plugin;
+		$this->media  = $plugin->get_component( 'media' );
+		$this->register_hooks();
+	}
+
+	/**
+	 * Register hooks.
+	 */
+	protected function register_hooks() {
+		add_filter( 'cloudinary_update_post_meta', array( $this, 'update_meta' ), 10, 4 );
+		add_filter( 'cloudinary_delete_post_meta', array( $this, 'delete_meta' ), 10, 3 );
+		add_filter( 'cloudinary_get_post_meta', array( $this, 'get_meta' ), 10, 3 );
+		add_action( 'cloudinary_upgrade_asset', array( $this, 'upgrade_relation' ), 10, 2 );
+	}
+
+	/**
+	 * Upgrade an asset relation.
+	 *
+	 * @param int    $attachment_id The attachment ID.
+	 * @param string $version       The version upgrading to.
+	 */
+	public function upgrade_relation( $attachment_id, $version ) {
+		$asset_plugin_version = $this->media->get_post_meta( $attachment_id, Sync::META_KEYS['plugin_version'], true );
+		if ( ! empty( $asset_plugin_version ) && version_compare( $asset_plugin_version, $version, '<' ) ) {
+			$meta = get_post_meta( $attachment_id, Sync::META_KEYS['cloudinary'], true );
+			if ( ! empty( $meta[ Sync::META_KEYS['transformation'] ] ) ) {
+				$this->media->update_post_meta( $attachment_id, Sync::META_KEYS['transformation'], $meta[ Sync::META_KEYS['transformation'] ] );
```

Comment:
@pereirinha right here it should unset the transformations and update the post meta.